src/modules/site-buyer/site-buyer.service.ts

```diff
@@ -0,0 +1,130 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import CustodianConfig from '@ukef/config/custodian.config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { UkefSiteId } from '@ukef/helpers';
+
+import { CustodianService } from '../custodian/custodian.service';
+import { CustodianCreateAndProvisionRequest } from '../custodian/dto/custodian-create-and-provision-request.dto';
+import { FieldEqualsListItemFilter } from '../sharepoint/list-item-filter/field-equals.list-item-filter';
+import { SharepointService } from '../sharepoint/sharepoint.service';
+import { CreateBuyerFolderRequestItem } from './dto/create-buyer-folder-request.dto';
+import { SiteExporterInvalidException } from './exception/site-exporter-invalid.exception';
+import { SiteExporterNotFoundException } from './exception/site-exporter-not-found.exception';
+
+type RequiredSharepointConfigKeys = 'scSharepointUrl' | 'scCaseSitesListId' | 'tfisSharepointUrl' | 'tfisCaseSitesListId' | 'scSiteFullUrl';
+type RequiredCustodianConfigKeys = 'buyerTemplateId' | 'buyerTypeGuid';
+
+@Injectable()
+export class SiteBuyerService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly sharepointConfig: Pick<ConfigType<typeof SharepointConfig>, RequiredSharepointConfigKeys>,
+    @Inject(CustodianConfig.KEY)
+    private readonly custodianConfig: Pick<ConfigType<typeof CustodianConfig>, RequiredCustodianConfigKeys>,
+    private readonly sharepointService: SharepointService,
+    private readonly custodianService: CustodianService,
+  ) {}
+
+  async createBuyerFolder(siteId: UkefSiteId, createBuyerFolderRequestItem: CreateBuyerFolderRequestItem): Promise<string> {
+    const { exporterName, buyerName } = createBuyerFolderRequestItem;
+
+    const caseSiteId = await this.getCaseSiteId(siteId);
+    const { buyerTermGuid, buyerUrl } = await this.getBuyerTermFromList(siteId, exporterName);
+
+    await this.sendCreateAndProvisionRequestForBuyerFolder(buyerName, caseSiteId, buyerTermGuid, buyerUrl);
+
+    return buyerName;
+  }
+
+  private async getCaseSiteId(siteId: string) {
+    const searchResults = await this.sharepointService.findListItems<{
+      id: string;
+      CustodianSiteURL: string;
+    }>({
+      siteUrl: this.sharepointConfig.scSharepointUrl,
+      listId: this.sharepointConfig.scCaseSitesListId,
+      fieldsToReturn: ['id', 'CustodianSiteURL'],
+      filter: new FieldEqualsListItemFilter({ fieldName: 'CustodianSiteURL', targetValue: siteId }),
+    });
+
+    if (searchResults.length === 0) {
```

Comment:
same as the other comment, we can simplify this into `if (!searchResults.length) { ... }`